flink-libraries/flink-table/src/main/scala/org/apache/flink/table/plan/ProjectionTranslator.scala

```diff
@@ -108,62 +109,73 @@ object ProjectionTranslator {
       tableEnv: TableEnvironment,
       aggNames: Map[Expression, String],
       propNames: Map[Expression, String]): Seq[NamedExpression] = {
-    exprs.map(replaceAggregationsAndProperties(_, tableEnv, aggNames, propNames))
-        .map(UnresolvedAlias)
-  }
 
-  private def replaceAggregationsAndProperties(
+    val projectNames: mutable.HashSet[String] = new mutable.HashSet[String]
+
+    def replaceAggregationsAndProperties(
```

Comment:
Can you rename this or the outer function? Having two functions with the same names is confusing.